images/linux/scripts/base/apt-mock.sh

```diff
@@ -14,15 +16,35 @@ while [ \$i -le 30 ];do
   err=\$(mktemp)
   $real_tool "\$@" 2>\$err
   result=\$?
-  if [ \$result -eq  0 ];then
-    break
+  cat \$err >&2
+
+  # no errors, continue
+  test \$result -eq  0 && continue
+
+  retry=false
+
+  if grep -q 'Could not get lock' \$err;then
+    # apt db locked needs retry
+    retry=true
+  elif grep -q 'Could not open file /var/lib/apt/lists' \$err;then
+    # apt update is not completed, needs retry
+    retry=true
+  elif grep -q 'IPC connect call failed' \$err;then
+    ## TODO:
+    ## the delay should help with gpg-agent not ready
+    ## if it is not then uncomment the folloing lines in order to
+    ## force restart gpg-agent
+    #  pkill -9 gpg-agent
+    #  sleep 1
+    #  source <(gpg-agent --daemon)
```

Comment:
can we remove this comment or is it still waiting to be tested?

Reply:
we should wait for the real life - it is not possible to be sure the simple delay solves the problem because it is not 100% reproducible

Reply:
I'd prefer to remove it because if your fix will work - these comments will live in the code for an uncertain amount of time. 
If the fix doesn't work - we can add it in additional PR.